packages/server/modules/core/tests/integration/user-emails.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { before } from 'mocha'
+import { createUser } from '../../services/users'
+import { beforeEachContext } from '@/test/hooks'
+import { expect } from 'chai'
+import { getUserByEmail } from '../../repositories/users'
+import { UserRecord } from '../../helpers/types'
+import crs from 'crypto-random-string'
+
+function createRandomEmail() {
+  return `${crs({ length: 6 })}@example.org`
+}
+
+describe('Core @user-emails', () => {
+  before(async () => {
+    await beforeEachContext()
+  })
+  describe('getUserByEmail', () => {
+    it('should return null if user does not exist', async () => {
+      expect(await getUserByEmail('test@example.org')).to.be.null
+    })
+
+    it('should return user if user-email does not exist', async () => {
+      const email = createRandomEmail()
+      await createUser({
+        name: 'John Doe',
+        email,
+        password: 'sn3aky-1337-b1m'
+      })
+      // TODO: delete user email
+
+      const user = (await getUserByEmail('test@example.org')) as UserRecord
```

Comment:
@alemagio Why the forced type cast at the end?

Reply:
https://www.notion.so/speckle/General-Web-Code-Review-Guide-1f2bf3dfea634460b3778ac3cf4aa1bb?pvs=4#75db5175359244e7bb988d753bfa07cd

Reply:
because of the nullable type, I don't see a big issue casts in tests

Reply:
use the non-null assertion operator then https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

at least in that case you're not overriding the entire type, only whether its null or not